### PR TITLE
Enable tests over jest

### PR DESCRIPTION
**Pull request by karelhala**
Simple example how to write tests and config which runs jest [1] is provided, this PR also removes webpack config specific to test and introduces new env variable ANALYZE, if such variable is given user may observer bunldes [2].

[1] https://facebook.github.io/jest/
[2] https://www.npmjs.com/package/webpack-bundle-analyzer

### CODE REVIEW

**Comment by Kinlaw** [comment]
@karelhala my bad, I let this slip and it has diverged quite a bit. Can you rebase?

**Comment by Kinlaw** [comment]
@jharting I think I have this cleaned up. Can you take a look when you get a chance?

**Comment by karelhala** [comment]
@Kinlaw looks good! I'd approve it, but I am original author of PR

File: .eslintignore [hunk 1]
```diff
@@ -1 +1,2 @@
-src/demoData/*
\ No newline at end of file
+src/demoData/*
+config/*
```
Comment (jharting) [inline .eslintignore]: I'd prefer config scripts linted as well

File: package.json [hunk 2]
```diff
@@ -68,13 +89,14 @@
...
     "lint": "npm-run-all lint:*",
     "lint:js": "eslint config src",
     "lint:sass": "stylelint 'src/**/*.scss' --config .stylelint.json",
     "prod": "WEBPACK_SERVE=production webpack-serve --config config/dev.webpack.config.js",
     "server:ctr": "node src/server/generateServerKey.js",
     "start": "WEBPACK_SERVE=development webpack-serve --config config/dev.webpack.config.js",
-    "test": "webpack --config config/test.webpack.config.js",
-    "verify": "npm-run-all build lint test"
+    "travis:test": "webpack --config config/test.webpack.config.js",
+    "verify": "npm-run-all build lint travis:test"
```
Comment (jharting) [inline package.json]: are tests not run in travis intentionally? Also I use `verify` as a shortcut for build + lint + test locally. Would be good to preserve its original meaning and define a new one solely for travis if we need one.

**Comment by Kinlaw** [comment]
@jharting fixed. Can you take another look?

File: config/base.webpack.plugins.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ plugins.push(CleanWebpackPlugin);
  * @type {var}
  */
 const LodashWebpackPlugin = new (require('lodash-webpack-plugin'))();
-plugins.push(LodashWebpackPlugin);
+process.env.ANALYZE && plugins.push(LodashWebpackPlugin);
```
Comment (jharting) [inline config/base.webpack.plugins.js]: how does this work?

Comment (Kinlaw) [reply config/base.webpack.plugins.js]: @jharting I'm honestly not sure. This was part of what @karelhala added. @karelhala is this still needed?

Comment (karelhala) [reply config/base.webpack.plugins.js]: This was probably a mistake introduced in my commit, I intended to anable `WebpackBundleAnalyzer` only when variable `ANALYZE` is present. However this should be wihtou process.env.ANALYZE.

And of course I forgot to submit my comment earlier, sory for that...

**Pull request merged**